api/src/main/java/org/openmrs/module/referencedemodata/patient/DemoPatientGenerator.java

```diff
@@ -317,9 +343,38 @@ private Visit createDemoVisit(Patient patient, List<VisitType> visitTypes, Locat
 				visit.addEncounter(createDemoLabsEncounter(patient, toDate(lastEncounterTime), admitLocation));
 			}
 			
+			if (randomDoubleBetween(0.0, 1.0) < .75) {
+				Encounter orderEncounter = null;
+				if (randomDoubleBetween(0.0, 1.0) < .5) {
+					orderEncounter = createDemoOrdersEncounter(patient, toDate(lastEncounterTime), location);
+					List<Concept> testConcepts = getConceptService().getConceptsByClass(getConceptService().getConceptClassByName("Test"));
+					if (testConcepts != null && testConcepts.size() > 0) {
+						createDemoOrder(orderEncounter, getOrderService().getOrderTypeByName("Test Order"), testConcepts);	
+					}
+				}
+				// TODO: create drug orders
+				
+				if (orderEncounter != null) {
+					visit.addEncounter(orderEncounter);
+				}
+			}
 			visit.addEncounter(createEncounter("Discharge", patient, toDate(dischargeTime), admitLocation));
 			visit.setStopDatetime(toDate(dischargeTime));
 		}
+		
+		visit = getVisitService().saveVisit(visit);
+		
+		if (randomDoubleBetween(0.0, 1.0) < .1) {
+			createDemoAppointment(patient, visit.getStartDatetime());
+		}
+		
+		if (randomDoubleBetween(0.0, 1.0) < .1) {
+			// there are three demo programs configured
+			List<Program> programs = getProgramWorkflowService().getAllPrograms(false);
+			if (programs != null && programs.size() > 0) {
+				createDemoPatientProgram(patient, programs.get(randomBetween(0, 30) % 3), visit.getStartDatetime());	
```

Comment:
There a reason not to use `randomArrayEntry()` here? (This implementation ties us pretty strongly to _only_ having 3 programs in the metadata plus the modulo actually biases things more than if this was just `randomBetween(0, 2)`).